internal/core/domain/link.go

```diff
@@ -0,0 +1,70 @@
+package domain
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	core "github.com/iden3/go-iden3-core"
+
+	"github.com/polygonid/sh-id-platform/internal/common"
+)
+
+// CredentialAttributes - credential's attributes
+type CredentialAttributes struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+// LinkCoreDID - represents a credential offer ID
+type LinkCoreDID core.DID
+
+// Link - represents a credential offer
+type Link struct {
+	ID                       uuid.UUID
+	IssuerDID                LinkCoreDID
+	CreatedAt                time.Time
+	MaxIssuance              *int
+	ValidUntil               *time.Time
+	SchemaID                 uuid.UUID
+	CredentialExpiration     *time.Time
+	CredentialSignatureProof bool
+	CredentialMTPProof       bool
+	CredentialAttributes     []CredentialAttributes
+	Active                   bool
+}
+
+// NewLink - Constructor
+func NewLink(id uuid.UUID, issuerDID core.DID, maxIssuance *int, validUntil *time.Time, schemaID uuid.UUID, credentialExpiration *time.Time, CredentialSignatureProof bool, CredentialMTPProof bool, credentialAttributes []CredentialAttributes) *Link {
```

Comment:
IMHO, The ID should get be initiated here, in the constructor.  Not as a parameter. This is the point where the Link entity is created, so this is a good point to give an ID.